scripts/linters/pylint_extensions_test.py

```diff
@@ -47,99 +47,98 @@ def test_no_break_after_hanging_indentation(self) -> None:
         node_break_after_hanging_indent = astroid.scoped_nodes.Module(
             name='test',
             doc='Custom test')
-        temp_file = tempfile.NamedTemporaryFile()
-        filename = temp_file.name
-        with utils.open_file(filename, 'w') as tmp:
-            tmp.write(
-                u"""self.post_json('/ml/\\trainedclassifierhandler',
-                self.payload, expect_errors=True, expected_status_int=401)
-                if (a > 1 and
-                        b > 2):
-                """)
-        node_break_after_hanging_indent.file = filename
-        node_break_after_hanging_indent.path = filename
-
-        self.checker_test_object.checker.process_tokens(
-           pylint_utils.tokenize_module(node_break_after_hanging_indent))
-
-        message = testutils.Message(
-            msg_id='no-break-after-hanging-indent', line=1)
-
-        with self.checker_test_object.assertAddsMessages(message):
-            temp_file.close()
+        with tempfile.NamedTemporaryFile() as temp_file:
+            filename = temp_file.name
+            with utils.open_file(filename, 'w') as tmp:
+                tmp.write(
+                    u"""self.post_json('/ml/\\trainedclassifierhandler',
+                    self.payload, expect_errors=True, expected_status_int=401)
+                    if (a > 1 and
+                            b > 2):
+                    """)
+            node_break_after_hanging_indent.file = filename
+            node_break_after_hanging_indent.path = filename
+
+            self.checker_test_object.checker.process_tokens(
+               pylint_utils.tokenize_module(node_break_after_hanging_indent))
+
+            message = testutils.Message(
+                msg_id='no-break-after-hanging-indent', line=1)
+
+            with self.checker_test_object.assertAddsMessages(message):
+                pass
```

Comment:
Hmm I'm worried by this. `self.checker_test_object.assertAddsMessages(message)` is a context manager asserting that inside its context, the message `message` is added. However, we aren't doing anything inside the context, so how can this work? I would expect to have to put `self.checker_test_object.checker.process_tokens(...` inside the context